frontends/ol-components/src/components/Button/Button.tsx

```diff
@@ -110,17 +110,22 @@ const ButtonStyled = styled.button<ButtonStyleProps>((props) => {
       backgroundColor: colors.mitRed,
       color: colors.white,
       border: "none",
+      /* Shadow/04dp */
+      boxShadow:
+        "0px 2px 4px 0px rgba(37, 38, 43, 0.10), 0px 3px 8px 0px rgba(37, 38, 43, 0.12)",
       ":hover:not(:disabled)": {
         backgroundColor: colors.red,
+        boxShadow: "none",
       },
       ":disabled": {
         backgroundColor: colors.silverGray,
+        boxShadow: "none",
       },
     },
     hasBorder && {
       backgroundColor: "transparent",
       borderColor: "currentcolor",
-      borderStyle: variant === "secondary" ? "solid" : "none",
```

Comment:
Previously, `const hasBorder = variant === "secondary" || variant === "text"`. But that was weird because `text` variant doesn't actually have a border. It was just set that way to pick up `backgroundColor: transparent`.

Applying `backgroundColor: transparent` to the text variant separately is simpler / more understandable, imo.